functional/unit_action_test.go

```diff
@@ -224,3 +316,139 @@ func TestUnitSSHActions(t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// read a sample unit file to a buffer
+	unitFile := "fixtures/units/hello.service"
+	fileBuf, err := ioutil.ReadFile(unitFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fileBody := strings.TrimSpace(string(fileBuf))
+
+	// submit a unit and assert it shows up
+	_, _, err = cluster.Fleetctl(m, "submit", unitFile)
+	if err != nil {
+		t.Fatalf("Unable to submit fleet unit: %v", err)
+	}
+	// wait until the unit gets submitted up to 15 seconds
+	_, err = runListUnits(cluster, m, 1)
+	if err != nil {
+		t.Fatalf("Failed to run list-units: %v", err)
+	}
+
+	// cat the unit file and compare it with the original unit body
+	stdout, _, err := cluster.Fleetctl(m, "cat", path.Base(unitFile))
+	if err != nil {
+		t.Fatalf("Unable to submit fleet unit: %v", err)
+	}
+	catBody := strings.TrimSpace(stdout)
+
+	if strings.Compare(catBody, fileBody) != 0 {
+		t.Fatalf("unit body changed across fleetctl cat: \noriginal:%s\nnew:%s", fileBody, catBody)
+	}
+}
+
+// TestUnitStatus simply checks "fleetctl status" actually works.
+func TestUnitStatus(t *testing.T) {
+	cluster, err := platform.NewNspawnCluster("smoke")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cluster.Destroy()
+
+	m, err := cluster.CreateMember()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = cluster.WaitForNMachines(m, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// submit a unit and assert it shows up
+	_, _, err = cluster.Fleetctl(m, "status")
+	if err != nil {
+		t.Fatalf("Unable to get fleet status: %v", err)
+	}
+}
+
+// runListUnits runs fleetctl list-units to verify the actual number of units
+// matched with the given expected number. It periodically runs list-units
+// waiting until list-units actually shows the expected units.
+func runListUnits(cluster platform.Cluster, m platform.Member, expectedUnits int) ([]string, error) {
+	units := make([]string, 0)
+	checkListUnits := func() bool {
+		outListUnits, _, err := cluster.Fleetctl(m, "list-units", "--no-legend")
+		if err != nil {
+			fmt.Errorf("Failed to run list-units: %v", err)
+			return false
+		}
+		if expectedUnits == 0 && len(outListUnits) == 0 {
+			return true
+		}
+		units = strings.Split(strings.TrimSpace(outListUnits), "\n")
+		if len(units) != expectedUnits {
+			// retry until number of units matched
+			fmt.Errorf("number mismatch %d, %d", len(units), expectedUnits)
+			return false
+		}
+		return true
+	}
+
+	_, err := util.WaitForState(checkListUnits)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to wait for checking list-units to finish: %v", err)
+	}
+
+	return units, nil
+}
+
+// runListUnitFiles runs fleetctl list-unit-files to verify the actual number
+// of units matched with the given expected number. It periodically runs
+// list-unit-files waiting until list-unit-files actually shows the expected
+// number of unit files.
+func runListUnitFiles(cluster platform.Cluster, m platform.Member, expectedUnits int) ([]string, error) {
```

Comment:
Same thing here, I've a functional test waiting #1515 that will use the runListUnitFiles() which is `fleetctl list-unit-files` and I want the returned output, so I could do the parsing. So WaitForNUnit() should be either inside these functions or just on its own. Functional tests should just be easy to write I want to run this function and get its output...